webserver/tyrus/src/main/java/io/helidon/webserver/tyrus/TyrusSupport.java

```diff
@@ -0,0 +1,300 @@
+        }
+        return DEFAULT_THREAD_POOL.get();
+    }
+
+    /**
+     * A Helidon handler that integrates with Tyrus and can process WebSocket
+     * upgrade requests.
+     */
+    private class TyrusHandler implements Handler {
+
+        /**
+         * Process a server request/response.
+         *
+         * @param req an HTTP server request.
+         * @param res an HTTP server response.
+         */
+        @Override
+        public void accept(ServerRequest req, ServerResponse res) {
+            // Skip this handler if not an upgrade request
+            Optional<String> secWebSocketKey = req.headers().value(HandshakeRequest.SEC_WEBSOCKET_KEY);
+            if (!secWebSocketKey.isPresent()) {
+                req.next();
+                return;
+            }
+
+            LOGGER.fine("Initiating WebSocket handshake ...");
+
+            // Create Tyrus request context and copy request headers
+            RequestContext requestContext = RequestContext.Builder.create()
+                    .requestURI(URI.create(req.path().toString()))      // excludes context path
+                    .build();
+            req.headers().toMap().entrySet().forEach(e ->
+                    requestContext.getHeaders().put(e.getKey(), e.getValue()));
+
+            // Use Tyrus to process a WebSocket upgrade request
+            final TyrusUpgradeResponse upgradeResponse = new TyrusUpgradeResponse();
+            final WebSocketEngine.UpgradeInfo upgradeInfo = engine.upgrade(requestContext, upgradeResponse);
+
+            // Respond to upgrade request using response from Tyrus
+            res.status(upgradeResponse.getStatus());
+            upgradeResponse.getHeaders().entrySet().forEach(e ->
+                    res.headers().add(e.getKey(), e.getValue()));
+            TyrusWriterPublisher publisherWriter = new TyrusWriterPublisher();
+            res.send(publisherWriter);
+
+            // Write reason for failure if not successful
+            if (upgradeInfo.getStatus() != WebSocketEngine.UpgradeStatus.SUCCESS) {
+                String reason = upgradeResponse.getReasonPhrase();
+                if (reason != null) {
+                    publisherWriter.write(ByteBuffer.wrap(reason.getBytes(UTF_8)), null);
+                }
+            }
+
+            // Flush upgrade response
+            publisherWriter.write(FLUSH_BUFFER, null);
+
+            // Setup the WebSocket connection and internally the ReaderHandler
+            Connection connection;
+            if (executorService != null) {
+                try {
+                    // Set up connection and call @onOpen
+                    Future<Connection> future = executorService.submit(() ->
+                            upgradeInfo.createConnection(publisherWriter,
+                                    closeReason -> LOGGER.fine(() -> "Connection closed: " + closeReason)));
+                    connection = future.get();      // Need to sync here
```

Comment:
I am not sure I understand this - why do you use an executor service, when you call `get` on the result? You are blocking the current thread anyways, why not just call the statement directly?
Also this seems to be an issue for our WebServer, as you should never block